### ISSUE TITLE
Failed import from snorkel.annotations: No module named templates

### ISSUE THREAD

**Issue by TamakiSakura** [HazyResearch/snorkel#622]
HI all, 

I was trying to run the `Intro_Tutorial_4` in the Jupiter environment initalized by `run.sh`, but I failed on the Step 6, with the following error:

```
ImportError                               Traceback (most recent call last)
<ipython-input-6-8b18ec4e65fe> in <module>()
----> 1 from snorkel.annotations import load_gold_labels
      2 L_gold_dev = load_gold_labels(session, annotator_name='gold', split=1)

/home/tamaki/Documents/Install/snorkel-master/snorkel/annotations.py in <module>()
      4 from sqlalchemy.sql import bindparam, select
      5 
----> 6 from .features import get_span_feats
      7 from .models import GoldLabel, GoldLabelKey, Label, LabelKey, Feature, FeatureKey, Candidate
      8 from .models.meta import new_sessionmaker

/home/tamaki/Documents/Install/snorkel-master/snorkel/features/__init__.py in <module>()
      7         get_sentence_token_count_feats,
      8 )
----> 9 from .entity_features import *
     10 from .generic_features import get_feats_from_matrix
     11 from .relative_features import (

/home/tamaki/Documents/Install/snorkel-master/snorkel/features/entity_features.py in <module>()
      1 import sys, os
      2 sys.path.append(os.environ['SNORKELHOME'] + '/treedlib/treedlib')
----> 3 from templates import *
      4 
      5 def compile_entity_feature_generator():

ImportError: No module named templates

```

I think I should had already installed all the required library, here is the last line of output I had when installing the dependencies as required by the installation guide

```
Successfully installed beautifulsoup4-4.6.0 bleach-2.0.0 bs4-0.0.1 configparser-
3.5.0 entrypoints-0.2.2 funcsigs-1.0.2 functools32-3.2.3.post2 futures-3.1.1 htm
l5lib-0.999999999 ipykernel-4.6.1 ipywidgets-6.0.0 jsonschema-2.6.0 jupyter-1.0.
0 jupyter-client-5.0.1 jupyter-console-5.1.0 jupyter-core-4.3.0 llvmlite-0.18.0 
lxml-3.6.4 mistune-0.7.4 mock-2.0.0 nbconvert-5.1.1 nbformat-4.3.0 nltk-3.2.3 no
tebook-5.0.0 numba-0.33.0 numbskull-0.0 numpy-1.12.1 pandocfilters-1.4.1 prompt-
toolkit-1.0.14 protobuf-3.3.0 ptyprocess-0.5.1 pyzmq-16.0.2 qtconsole-4.3.0 scip
y-0.19.0 sqlalchemy-1.1.9 tensorflow-1.1.0 terminado-0.6 testpath-0.3 tika-1.14.
1 traitlets-4.3.2 wcwidth-0.1.7 webencodings-0.5.1 werkzeug-0.12.2 widgetsnbexte
nsion-2.0.0
```

Sorry if this is a silly question.

**Comment by TamakiSakura** [comment]
My bad - If I download the the snorkel from the source, the treelib doesn't come with it, I have to download it separately.

**Comment by ajratner** [comment]
Hm, the `run.sh` should have taken care of this for you... we'll look into this further!

**Comment by TamakiSakura** [comment]
It has, I realize the reason why I had this problem is I download the source as zip, so the `git submodule update --init --recursive` didn't recognize it's a git directory and treelib has to be updated.

**Comment by ajratner** [comment]
Ok great, thanks for writing back!

**Comment by camilix** [comment]
Hi may I ask how is this issue solve, because I just got the same ImportError

**Comment by TamakiSakura** [comment]
@camilix Download https://github.com/HazyResearch/treedlib/tree/b0295695d1eea5596bd5619447569ee159d4b21e and put it in the snorkel directory.

**Comment by zkawan** [comment]
Hi, I came across the same error in Intro_Tutorial_2 and followed what you have suggested, i.e. downloaded treedlib and placed in snorkel directory but the error persists in step 3

**Comment by frhino** [comment]
I'm running Windows 10. I first tried @TamakiSakura's suggestion, but I've never installed a module in that way, so wasn't sure of any of the specifics, so of course I failed (no fault of @TamakiSakura ). But then, in my anaconda prompt (running as admin, but not sure that matters) while in snorkel environment, I just did <code>pip install treedlib</code> and then relaunched jupyter notebook and was able to get past that error. In the next couple of cells, I stumbled on an error missing the numbskull module, and after staring at that for a second, I pip installed that the same way and all was good for the rest of the notebook.
